examples/OcclusionSaliency.ipynb

```diff
@@ -102,38 +119,56 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Input image loader to match feature extractor expectation\n",
-    "imsize = 224\n",
-    "loader = transforms.Compose([transforms.Resize((imsize)), transforms.ToTensor()])\n",
-    "\n",
-    "def image_loader(image_name):\n",
-    "    try:\n",
-    "        image = Image.open(image_name)\n",
-    "    except AttributeError:\n",
-    "        # If Image already open\n",
-    "        image = image_name\n",
-    "    image = loader(image).float()\n",
-    "    image = Variable(image, requires_grad=False)\n",
-    "    return image.cuda().unsqueeze(0)\n",
+    "# Setup our XAI components for image perturbation and saliency map\n",
+    "# generation from classification black-box results.\n",
+    "from smqtk_core.configuration import from_config_dict\n",
+    "from xaitk_saliency import PerturbImage, ImageClassifierSaliencyMapGenerator\n",
+    "from xaitk_saliency.impls.perturb_image.sliding_window import SlidingWindowPerturb\n",
+    "from xaitk_saliency.impls.perturb_image.rise import RISEPertubation\n",
+    "from xaitk_saliency.impls.vis_sal_classifier.occlusion_scoring import OcclusionScoring\n",
     "\n",
-    "def predict_with_model(model, image_tensor):\n",
-    "    featureVec = model(image_tensor)\n",
-    "    # Converting feature extractor output to probabilities.\n",
-    "    preds_ = torch.nn.functional.softmax(featureVec, dim=1).cpu().detach().numpy().squeeze()\n",
-    "    # Returning scores for only desired class IDs\n",
-    "    return np.asarray(itemgetter(*required_class_id)(preds_))\n",
+    "perturber = SlidingWindowPerturb(\n",
+    "    window_size=(24, 24),\n",
+    "    stride=(4, 4),   # evenly divides the window size per-axis.\n",
+    ")\n",
+    "# perturber = RISEPertubation(\n",
```

Comment:
I'm wondering if it's better practice to actually run both algorithms (sliding window and RISE) and compare them side-by-side. One of the benefits of notebooks is that we can visualize the output results of both.

Reply:
The original intention was to show an "application" like work flow where one is the chosen path. In the demo that used this notebook, we shows that we could change the part of that once cell that sets the perturber, by direct assignment or by configuration, to something different without impacting the rest of the "application."

Reply:
I have pushed an additional commit (for ease of rollback purposes) that attempts a revision of the framing and narrative of the notebook. Please take a look and see if this better shows what we want here. When we settle I will squash the experiments here to a single commit as the unified "change."

Reply:
Looks good to me- I added a minor comment about maybe not using auto-normalization when showing mask coverage, since it amplifies small differences and puts sliding window and RISE on two different axes. Thanks!